src/ast/passes/resource_analyser.cpp

```diff
@@ -124,6 +131,12 @@ void ResourceAnalyser::visit(Builtin &builtin)
     // and symbols resolved even when unavailable at resolution time
     resources_.probes_using_usym.insert(probe_);
   }
+  if (builtin.ident == "probe") {
+    if (exceeds_stack_limit(builtin.type.GetSize())) {
+      resources_.probe_str_buffers = 1;
```

Comment:
I was thinking that because there can be only one `probe` builtin per program and one program per cpu that it would be ok but maybe there is interleaving that could affect this? I'm not sure so I'll just increment as suggested.

Reply:
Ah, I see, that's probably a good point. I'd have to think about it a bit more, though.